core/payment/src/processor.rs

```diff
@@ -1,203 +1,288 @@
+        match self.drivers.lock().await.entry(msg.driver.clone()) {
+            Entry::Vacant(entry) => {
+                entry.insert(msg.platform.clone());
+            }
+            Entry::Occupied(entry) if entry.get() != &msg.platform => {
+                return Err(RegisterAccountError::Other(format!(
+                    "Driver {} is registered as handling platform {}",
+                    msg.driver,
+                    entry.get()
+                )))
             }
+            _ => {}
         }
+
+        match self
+            .accounts
+            .lock()
+            .await
+            .entry((msg.platform, msg.address))
+        {
+            Entry::Occupied(_) => return Err(RegisterAccountError::AlreadyRegistered),
+            Entry::Vacant(entry) => entry.insert((msg.driver, msg.mode)),
+        };
+        Ok(())
     }
 
-    async fn process_payment(&self, msg: SchedulePayment) {
-        let payer_id = msg.payer_id;
-        let payee_id = msg.payee_id;
-        let payer_addr = msg.payer_addr.clone();
-        let payee_addr = msg.payee_addr.clone();
-
-        let result: Result<(), Error> = async move {
-            let confirmation = self.wait_for_payment(&msg.document_id()).await?;
-
-            let (activity_payments, agreement_payments) = match &msg.title {
-                PaymentTitle::DebitNote(debit_note_payment) => (
-                    vec![ActivityPayment {
-                        activity_id: debit_note_payment.activity_id.clone(),
-                        amount: msg.amount.clone(),
-                    }],
-                    vec![],
-                ),
-                PaymentTitle::Invoice(invoice_payment) => (
-                    vec![],
-                    vec![AgreementPayment {
-                        agreement_id: invoice_payment.agreement_id.clone(),
-                        amount: msg.amount.clone(),
-                    }],
-                ),
-            };
-
-            let payment_dao: PaymentDao = self.db_executor.as_dao();
-            let payment_id = payment_dao
-                .create_new(
-                    payer_id,
-                    payee_id,
-                    payer_addr,
-                    payee_addr,
-                    msg.allocation_id,
-                    msg.amount,
-                    confirmation.confirmation,
-                    activity_payments,
-                    agreement_payments,
-                )
-                .await?;
-            let payment = payment_dao.get(payment_id, payer_id).await?.unwrap();
-            let payee_id = payment.payee_id;
-            let msg = SendPayment(payment);
-            ya_net::from(payer_id)
-                .to(payee_id)
-                .service(BUS_ID)
-                .call(msg)
-                .await??;
-
-            Ok(())
+    pub async fn unregister_account(
+        &self,
+        msg: UnregisterAccount,
+    ) -> Result<(), UnregisterAccountError> {
+        match self
```

Comment:
Yes, because the idea behind `drivers` is to have a mapping showing which driver handles which platform. When an account is unregistered, we don't want to remove the driver from that mapping (there might still be other accounts registered by this driver).

Reply:
Ahh.. okay